Server.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+#include <boost/asio.hpp>
+#include <iostream>
+#include <memory>
+#include <array>
+#include <vector>
+#include <unordered_map>
+
+
+class Server
+{
+
+private:
+	static constexpr auto PORT = 8080;
+	boost::asio::io_service _io_service;
+	boost::asio::ip::tcp::acceptor _acceptor;
+	boost::asio::ip::tcp::socket _socket;
+	std::unordered_map<int, std::shared_ptr<boost::asio::ip::tcp::socket>> _clients;
+	std::shared_ptr<std::vector<char>> _bufferptr = std::make_shared<std::vector<char>>();
```

Comment:
initialize this in the constrator

Reply:
why even wrap the buffer in a `shared_ptr` in the first place